### DISCUSSION TITLE
Uvicorn sometimes doesn't start server process

### DISCUSSION THREAD

**Discussion by arvid123** [Kludex/uvicorn#2083] [Potential Issue]
Sometimes, seemingly randomly uvicorn gets stuck after starting the reloader process when I start uvicorn using this command:

```bash
poetry run uvicorn --host 0.0.0.0 --port 8000 --reload lambert.api.main:app
```

Or with this vscode launch configuration

```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Lambert",
            "type": "python",
            "request": "launch",
            "module": "poetry",
            "args": [
                "run",
                "python",
                "-m",
                "uvicorn",
                "lambert.api.main:app",
                "--reload",
                "--host",
                "localhost",
                "--port",
                "8000"
            ],
            "jinja": true,
            "justMyCode": true,
        }
    ]
}
```

This is the output:

```
INFO:     Will watch for changes in these directories: ['/home/arvid/leya/lambert']
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [16920] using StatReload
```

I believe that it might not get completely stuck, it might just be very delayed.

Lambert is the name of the app btw.

We are using uvicorn together with FastAPI and i believe that this error started occuring after we started using websockets.
Has anyone had a similar issue or know of a solution?

Uvicorn version: 0.21.1
FastAPI version: 0.103.0

**Comment by Kludex** [comment]
Do you have an MRE?

**Reply by arvid123** [reply]
What is an MRE?

**Reply by Kludex** [reply]
https://en.wikipedia.org/wiki/Minimal_reproducible_example

**Reply by arvid123** [reply]
I have tried but not been able to reproduce it in a small test project

**Reply by Kludex** [reply]
I don't know if it's related: https://github.com/tiangolo/fastapi/discussions/9149.

Without a MRE, I'll not be able to help.
